src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        if self.config.output_router_logits or output_router_logits:
+            ret["router_logits"] = out_router_logits
+        ret["loss"] = loss
+
+        if return_dict:
+            return ret
+
+        outp = collections.namedtuple("GPTSANOutputs", " ".join(ret.keys()))
+        return outp(**ret)
+
+    def prepare_inputs_for_generation(
+        self,
+        input_ids: torch.LongTensor,
+        connected_inputs: Optional[torch.LongTensor] = None,
+        spout: Optional[torch.FloatTensor] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+        random_seed: Optional[int] = None,
+        **kwargs
+    ):
+        if past_key_values is not None:
+            return {
+                "input_ids": input_ids,
+                "num_precontext": connected_inputs,
+                "spout": None,
+                "past_key_values": past_key_values,
+            }
+        NUM_TOKENS = self.config.vocab_size
+        SOT_TOKEN = NUM_TOKENS - 7
+        SEG_TOKEN = NUM_TOKENS - 2
+        if connected_inputs is None:
+            connected_inputs = torch.zeros(input_ids.shape[0]).int().to(input_ids.device)
+        pre_input = torch.stack(
+            [
+                torch.tensor(
+                    [SOT_TOKEN]
+                    + input_ids[i, : connected_inputs[i]].cpu().numpy().tolist()
+                    + [SEG_TOKEN]
+                    + input_ids[i, connected_inputs[i] :].cpu().numpy().tolist()
+                )
+                for i in range(input_ids.shape[0])
+            ]
+        )
+        pre_input = pre_input.to(input_ids.device)
+        connected_inputs = connected_inputs + 1
+        connected_inputs = torch.unsqueeze(connected_inputs, dim=1)
+        if random_seed is not None:
+            # PyTorchの内部を決定論的に設定する
+            torch.backends.cudnn.deterministic = True
+            torch.backends.cudnn.benchmark = False
+            np.random.seed(random_seed)
+            torch.manual_seed(random_seed)
+        return {
+            "input_ids": pre_input,
+            "num_precontext": connected_inputs,
+            "spout": spout,
+            "past_key_values": past_key_values,
+        }
+
+    def prepare_decoder_input_ids_from_labels(self, labels: torch.Tensor):
+        return labels
+
+    def _get_logits_processor(
+        self,
+        generation_config: GenerationConfig,
+        input_ids_seq_length: int,
+        encoder_input_ids: torch.LongTensor,
+        prefix_allowed_tokens_fn: Callable[[int, torch.Tensor], List[int]],
+        logits_processor: Optional[LogitsProcessorList],
+    ) -> LogitsProcessorList:
+        logits_processor = super()._get_logits_processor(
+            generation_config, input_ids_seq_length, encoder_input_ids, prefix_allowed_tokens_fn, logits_processor
+        )
+        if generation_config.top_k is not None:
+            logits_processor.append(TopKLogitsWarper(generation_config.top_k))
+        return logits_processor
```

Comment:
This is not really needed. It is automatically added [here](https://github.com/ArthurZucker/transformers/blob/7547c82c2f0768cc7e7034e445ce05dc743e6c0c/src/transformers/generation/utils.py#L1429)